packages/advanced-logic/src/extensions/payment-network/declarative.ts

```diff
@@ -347,11 +361,14 @@ export default class DeclarativePaymentNetwork<
     actionSigner: IdentityTypes.IIdentity,
     timestamp: number,
   ): ExtensionTypes.IState {
-    if (!requestState.payer) {
-      throw Error(`The request must have a payer`);
+    if (!requestState.payer && !extensionState.values.thirdparty) {
+      throw Error(`The request must have a payer or a thirdparty`);
     }
-    if (!Utils.identity.areEqual(actionSigner, requestState.payer)) {
-      throw Error(`The signer must be the payer`);
+    if (
+      !Utils.identity.areEqual(actionSigner, requestState.payer) &&
+      !Utils.identity.areEqual(actionSigner, extensionState.values.thirdparty)
+    ) {
+      throw Error(`The signer must be the payer or the thirdparty`);
```

Comment:
I don't think the third-party should have payer privileges. Maybe there should be a payee third party and a payer third party (in a later version ?) but one actor can only delegate his own action to a third-party, in my opinion.

Let's take bank accounts, if a third-party appointed by the issuer can make sure the refunds has been sent, it cannot know on behalf of the payer that the money has been received.

(this comment applies for all the functions below where the third party can impersonate the payer)

Reply:
this is in line with my earlier comment about multiple thirdparties

Reply:
OK, are we in line also to make the multiple third parties in another release ?